### PR TITLE
Update to benefit from Go 1.17

### PR DESCRIPTION
**Pull request by mtrmac**
Per https://github.com/containers/image/pull/1570#issuecomment-1149876963 , Go 1.17 will be required; so, update c/image to benefit from it.

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
LGTM

File: pkg/docker/config/config_test.go [hunk 1]
```diff
@@ -61,40 +50,35 @@ func TestGetPathToAuth(t *testing.T) {
...
-		if c.expected == "" {
-			assert.Error(t, err)
-		} else {
-			require.NoError(t, err)
-			assert.Equal(t, c.expected, res)
-			assert.Equal(t, c.legacyFormat, lf)
-		}
+		t.Run(fmt.Sprintf("%d", caseIndex), func(t *testing.T) {
+			t.Setenv("XDG_RUNTIME_DIR", c.xrd)
+			if c.xrd == "" {
```
Comment (rhatdan) [inline pkg/docker/config/config_test.go:55]: This seems a little weird, why not only set XDG_RUNTIME_DIR if c.xrd is set? Unless you are worried about XDG_RUNTIME_DIR already being set.

Comment (mtrmac) [reply pkg/docker/config/config_test.go:58]: Thanks for pointing this out, this really deserves a comment. Updated now, in both places.

(`t.Setenv`, unlike `os.Setenv`, ensures the environment variable is always restored to the original value (or original unset state) after the test finishes; and there is no `t.Unsetenv`. So we call `t.Setenv` unconditionally to get get the cleanup guarantee, and only then worry about possibly unsetting the variable if wanted by the test.)

**Comment by rhatdan** [comment]
LGTM

**Pull request merged**